src/shared/guards/environment.guard.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common'
+import { Observable } from 'rxjs'
+
+
+// Used to disable endpoints in production
+@Injectable()
+export class ENVGuard implements CanActivate {
+    canActivate(
+        context: ExecutionContext,
+    ): boolean | Promise<boolean> | Observable<boolean> {
```

Comment:
Only boolean return type?

Reply:
This is how the Nest Docs do it...so it may be needed for under the hood operations

(https://docs.nestjs.com/guards)